### DISCUSSION TITLE
Request for Documentation on AdminerEvo Editor

### DISCUSSION THREAD

**Discussion by LionelLaffineur** [adminerevo/adminerevo#57] [General]
Dear all,

Right now, we miss documentation about AdminerEvo Editor.  This is quite important to help deploy and start using it easily.
We are interested in someone to help writing about it.  It should ideally be written in Markdown format to be easily integrated in existing documentation.  It's not mandatory to provide it via pull request, file(s) can be attached to this discussion and someone of the team will take care of installing them at the right place in the project.

We thank you already in advance,
L.

**Comment by aoirthoir** [comment]
I am interested. The problem of course is, that there is no documentation that gives me a starting point. BUT point me in the write direction and I am game to help.

I know PHP from way back but I am not an OOP guy, so the code base may take some time for me to really understand what is going on. To be clear, I started with COBOL back in the day, still use it occasionally know so when I use PHP i do things like:

```php
if ($a > $b):
    echo $a." is greater than ".$b;
elseif ($a == $b): // Note the combination of the words.
    echo $a." equals ".$b;
else:
    echo $a." is neither greater than or equal to ".$b;
endif;
```

So of course I can read typically PHP with curly braces but I am unskilled in OOP and continue after all these years to see the point. I dont say this to argue about the usage of OOP. I believe adminerevo uses oop and its a great product. I just say this to preface that I am willing to work on the markdown documentation for editor, as I would like it to be a viable product, but I may have questions along the way.

**Reply by LionelLaffineur** [reply]
Dear @aoirthoir, you are for sure welcome to write something !
It is true that there is not really something to point you too.  And it's not tomorrow that I will have time to write something since the list of topics is growing faster than we can process !
What is really needed is a beginner's guide; if I were you, I'd setup a fresh install somewhere and take screenshots all along to make it a beginner's guide.
Regards,
L.

**Comment by aoirthoir** [comment]
I will try with your version. The problem was that when I tried Editor with the old version, since there were not instructions, I had no idea what to do.

**Comment by d-roman-halliday** [comment]
Hi, I'd be glad to contribute some documentation (I'm working on my own notes). But I can't seem to get fully started myself. Is there much more documentation/examples around?

For the documentation, where would you like me to add my Markdown? I'm happy to put it in a separate readme under the editor directory, or in the main one.

I've setup adminer without too much difficulty (for Postgres and MySQL), the editor is less straightforward. 

For the editor, it seems to be a requirement to build my own page/application on top of the existing code (using the extension approach)? The best information/instructions I can find are:
https://www.adminer.org/en/extension/

Is recreating the `loginForm()` the best/right way to manage connection settings for the database (particularly postgres), where both `auth[server]` and `auth[driver]` need to be defined? Or is there a better way of doing this, like with `database()`?

### MySQL
Basic editor `editor.php`, it tells me:
> No database selected

Creating my own page/application extending the core functionality (as below) seems to work for MySQL:
```
<?php
function adminer_object() {

  class AdminerSoftware extends Adminer {

    function name() {
      // custom name in title and heading
      return 'MySQL Test Adminer Editor';
    }

    function database() {
      // database name, will be escaped by Adminer
      return '<database_name>';
    }

  }

  return new AdminerSoftware;
}

include './editor.php';
```

### Postgres
When using `editor-pgsql.php` I get the login prompt, and I put in the username/password... Whatever I enter, I find myself just at the login prompt again (although it now lists what I entered as a username at the side). 

```
<?php
function adminer_object() {
  
  class AdminerSoftware extends Adminer {

    function name() {
      // custom name in title and heading
      return 'Postgres Test Adminer Editor';
    }
    
    function database() {
      // database name, will be escaped by Adminer
      return '<database_name>';
    }
    
    function loginForm() {
      echo "<table cellspacing='0' class='layout'>\n";
      echo $this->loginFormField('username', '<tr><th>' . lang('Username') . '<td>', '<input type="hidden" name="auth[driver]" value="pgsql"><input type="hidden" name="auth[server]" value="127.0.0.1"><input name="auth[username]" id="username" value="' . h($_GET["username"]) . '" autocomplete="username" autocapitalize="off">' . script("focus(qs('#username'));"));
      echo $this->loginFormField('password', '<tr><th>' . lang('Password') . '<td>', '<input type="password" name="auth[password]" autocomplete="current-password">' . "\n");
      echo "</table>\n";
      echo "<p><input type='submit' value='" . lang('Login') . "'>\n";
      echo checkbox("auth[permanent]", 1, $_COOKIE["adminer_permanent"], lang('Permanent login')) . "\n";
    }
    
  }
  
  return new AdminerSoftware;
}

include './editor-pgsql.php';
```

**Comment by LionelLaffineur** [comment]
I am not sure about the right way to go for documentation.
I was thinking at creating a set of .md files under /docs in sub-folders that would do the categorization: i.e. how-to.
But how to make sure anyone who wants to write documentation will use a standard generic layout the same for all documents ?

**Comment by mckaygerhard** [comment]
please i can handle all the task but the best way is to create a spoecific repo for! @LionelLaffineur  .. i was thinking about enabling the wiki part of the repo but a repository with a bunch of MD file is fully compatible with almost any modern wiki like wiki.js by example
